### PR TITLE
Implement the ability to pass a block to encode_io and encode

### LINKED ISSUE #34: Loop over all emoji in string
Is it possible to loop over all the emoji in the string?

Example possible api:

    > Rumoji.encode_each("🐻  regular text ♥️") { |emoji| "<<#{emoji.name}>>" }
    > "<<bear>>  regular text <<hearts>>"

This would allow me to customize the display of emoji, converting them into images or stripping out undesirable ones. Ideally this would also work with mutli-character emoji like 👪🏾 as a single emoji: http://emojipedia.org/family-type-5/

### PR DESCRIPTION
**Pull request by axelson**
This allows a library user to easily customize the representation of an
emoji (mapping it to a custom image for example)

Note: this exposes the Emoji object to the end-user (which can be very useful)

This PR isn't complete but I wanted to get the core of the code up to get some initial feedback

TODO:
- [x] docs
- [x] tests

Fixes #34

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +55,37 @@ On the command line
...
+#### .code
+
+The symbol of the emoji surrounded with colons
+
+    Rumoji.encode("😭") {|emoji| emoji.code}
+    #=> ":sob:"
+
+#### .symbol
+
+The symbol of the emoji surrounded with colons
```
Comment (mwunsch) [inline README.md]: Typo here, I think

File: lib/rumoji.rb [hunk 2]
```diff
@@ -8,17 +8,25 @@ module Rumoji
...
-    str.gsub(Emoji::ALL_REGEXP) { |match| (emoji = Emoji.find_by_string(match)) && emoji.code || match }
+    str.gsub(Emoji::ALL_REGEXP) do |match|
+      if emoji = Emoji.find_by_string(match)
+        if block_given?
+          yield emoji
+        else
+          emoji.code || match
+        end
+      end
+    end
```
Comment (mwunsch) [inline lib/rumoji.rb]: The outer if block needs an else clause, that's where the `match` goes (if an emoji is not present return the matched text).

File: README.md (hunk 1 above)
Comment (axelson) [inline README.md]: fixed

File: lib/rumoji.rb (hunk 2 above)
Comment (axelson) [inline lib/rumoji.rb]: Ah, true. Nice catch. Tried to write a failing test for it but I'm not sure how to construct a string with an emoji that would match `ALL_REGEXP` but not `STRING_LOOKUP`.

**Comment by axelson** [comment]
Okay, updated with changes

**Comment by mwunsch** [comment]
Nicely done!

**Comment by axelson** [comment]
Thanks! 👍

**Pull request merged**